Sources/StreamChatUI/ChatChannelList/ChatChannelListVC_Tests.swift

```diff
@@ -13,6 +13,47 @@ class ChatChannelListVC_Tests: XCTestCase {
     var mockedChannelListController: ChatChannelListController_Mock<NoExtraData>!
     var mockedCurrentUserController: CurrentChatUserController_Mock<NoExtraData>!
     var mockedRouter: ChatChannelListRouter_Mock<NoExtraData> { vc.router as! ChatChannelListRouter_Mock<NoExtraData> }
+
+    var mockedChannels: [ChatChannel] {
+        let channel = ChatChannel.mock(
+            cid: .unique,
+            name: "Channel 1",
+            imageURL: TestImages.yoda.url,
+            lastMessageAt: .init(timeIntervalSince1970: 1_611_951_526_000)
+        )
+        let channelWithOnlineIndicator = ChatChannel.mockDMChannel(
+            lastMessageAt: .init(timeIntervalSince1970: 1_611_951_527_000),
+            members: [.mock(id: .unique, name: "Darth Vader", imageURL: TestImages.vader.url, isOnline: true)]
+        )
+        let channelWithLongTextAndUnreadCount = ChatChannel.mock(
+            cid: .init(type: .messaging, id: "test_channel3"),
+            name: "This is a channel with a big name. Really big.",
+            imageURL: TestImages.yoda.url,
+            lastMessageAt: .init(timeIntervalSince1970: 1_611_951_528_000),
+            unreadCount: .mock(messages: 4),
+            latestMessages: [
+                ChatMessage.mock(
+                    id: "1", text: "This is a long message. How the UI will adjust?", author: .mock(id: "Vader2")
+                )
+            ]
+        )
+        let channelWithMultipleMessages = ChatChannel.mock(
+            cid: .init(type: .messaging, id: "test_channel4"),
+            name: "Channel 4",
+            imageURL: TestImages.vader.url,
+            lastMessageAt: .init(timeIntervalSince1970: 1_611_951_529_000),
+            latestMessages: [
+                ChatMessage.mock(id: "2", text: "Hello", author: .mock(id: "Vader")),
+                ChatMessage.mock(id: "1", text: "Hello2", author: .mock(id: "Vader2"))
+            ]
+        )
+        return [
+            channel,
+            channelWithOnlineIndicator,
+            channelWithLongTextAndUnreadCount,
+            channelWithMultipleMessages
+        ]
+    }
```

Comment:
This is a computed property which means that every time you access it, a new set of channels (some of them with a unique id every time) is created.

I kinda like the idea of having a set of various channels ready to use for tests. Maybe it would make sense to put this code into some kind of `dummy` helper (like we do it for LLC tests) and then just assign the property local property in `setUp()` as we do it everywhere else.

I already commented on a similar this here https://github.com/GetStream/stream-chat-swift/pull/853#discussion_r585358400 . Please get used to the fact that unless you _really want_ the variable to be constant (i.e. snapshot size), it better to set it in `setUp`. It will save you headaches in the future.

Reply:
I don't think the dummy helper here is very beneficial. Because it is not dummy data, and it is useful to see in this place all the data that will be displayed in the snapshot. But the setup of the data has been changed 👍